addons/stock/views/product_views.xml

```diff
@@ -485,100 +504,100 @@
-            </p>
-        </field>
-    </record>
+        <record id="product_template_action_product" model="ir.actions.act_window">
+            <field name="name">Products</field>
+            <field name="res_model">product.template</field>
+            <field name="view_mode">kanban,tree,form</field>
+            <field name="search_view_id" ref="product_template_search_form_view_stock"/>
+            <field name="context">{"search_default_consumable": 1, 'default_detailed_type': 'product'}</field>
+            <field name="help" type="html">
+                <p class="o_view_nocontent_smiling_face">
+                    No product found. Let's create one!
+                </p><p>
+                    Track your stock quantities by creating storable products.
+                </p>
+            </field>
+        </record>
 
-    <record id="stock_product_normal_action" model="ir.actions.act_window">
-        <field name="name">Product Variants</field>
-        <field name="res_model">product.product</field>
-        <field name="view_mode">tree,form,kanban</field>
-        <field name="search_view_id" ref="stock_product_search_form_view"/>
-    </record>
+        <record id="stock_product_normal_action" model="ir.actions.act_window">
+            <field name="name">Product Variants</field>
+            <field name="res_model">product.product</field>
+            <field name="view_mode">tree,form,kanban</field>
+            <field name="search_view_id" ref="stock_product_search_form_view"/>
+        </record>
 
-    <menuitem id="menu_product_variant_config_stock" name="Products" action="product_template_action_product"
-        parent="stock.menu_stock_inventory_control" sequence="1"/>
-    <menuitem id="product_product_menu" name="Product Variants" action="stock_product_normal_action"
-        parent="menu_stock_inventory_control" sequence="2" groups="product.group_product_variant"/>
-    <menuitem id="menu_product_packagings" name="Product Packagings" parent="stock.menu_product_in_config_stock" action="product.action_packaging_view" groups="product.group_stock_packaging"/>
-    <menuitem id="menu_product_stock" name="Stock" action="stock.action_product_stock_view"
+        <menuitem id="menu_product_variant_config_stock" name="Products" action="product_template_action_product"
+            parent="stock.menu_stock_inventory_control" sequence="1"/>
+        <menuitem id="product_product_menu" name="Product Variants" action="stock_product_normal_action"
+            parent="menu_stock_inventory_control" sequence="2" groups="product.group_product_variant"/>
+        <menuitem id="menu_product_packagings" name="Product Packagings" parent="stock.menu_product_in_config_stock" action="product.action_packaging_view" groups="product.group_stock_packaging"/>
+        <menuitem id="menu_product_stock" name="Stock" action="stock.action_product_stock_view"
         parent="stock.menu_warehouse_report" sequence="5"/>
```

Comment:
If you intend to clean a bit around in this PR, I'd recommend defining a `stock_menus.xml` file, using the recursive definition of menuitems (cf `sale_menus.xml` or other files like that).
It's way easier to read and understand ;)